mypy/server/astdiff.py

```diff
@@ -335,9 +335,6 @@ def visit_union_type(self, typ: UnionType) -> SnapshotItem:
         normalized = tuple(sorted(items))
         return ('UnionType', normalized)
 
-    def visit_type_guard_type(self, typ: TypeGuardType) -> SnapshotItem:
-        return ('TypeGuardType', snapshot_type(typ.type_guard))
```

Comment:
I think this may be wrong. We still need to have some way to propagate the information on updated type guard function. (Btw most likely the old way was wrong as well, do we actually have good test cover for type guards in daemon mode?)